code/android-signal-library/src/main/java/com/adobe/marketing/mobile/signal/SignalExtension.kt

```diff
@@ -0,0 +1,148 @@
+/*
+  Copyright 2022 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+ */
+package com.adobe.marketing.mobile.signal
+
+import com.adobe.marketing.mobile.*
+import com.adobe.marketing.mobile.services.HitQueuing
+import com.adobe.marketing.mobile.services.PersistentHitQueue
+import com.adobe.marketing.mobile.services.ServiceProvider
+import com.adobe.marketing.mobile.utils.DataReader
+
+class SignalExtension(extensionApi: ExtensionApi?) : Extension(extensionApi) {
```

Comment:
Any reason why `ExtensionApi` is an optional?

Reply:
https://github.com/adobe/aepsdk-core-android/blob/258d31272a6dd41f3fdacd2c6045297f93db6a64/code/android-core-library/src/main/java/com/adobe/marketing/mobile/Extension.java#L28
As we discussed before, we will not add `NotNull` annotation to `Extension` class for backward compatibility. So, technically the Kotlin constructor could take in null.

Reply:
Considering the constructor is `protected` is there a possibility of getting a `null` value? @prudrabhat might need to do the same in configuration as well

Reply:
This won't be non null. I have also added `NonNull` annotation to Extension class to force kotlin implementations to use non null type.